File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/entities/notification.js


import { Entity } from './entity.js';
class Notification extends Entity{
    constructor() {
        super();
        this.NotificationId = null;
        this.UserId = null;
        this.Thumbnail = null;
        this.Title = null;
        this.Content = null;
        this.ReferenceLink = null;
        this.Time = null;
        this.IsRead = false;
    }

    init() {
        return new Notification();
    }
}

export default Notification;

File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Entities/Notification.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Entities
{
    public class Notification : Entity
    {
        [Key]
        public Guid NotificationId { get; set; } = Guid.Empty;
        public Guid UserId { get; set; } = Guid.Empty;
        public string? Thumbnail { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string ReferenceLink { get; set; } = string.Empty;
        public DateTime Time { get; set; } = DateTime.UtcNow;
        public bool IsRead { get; set; } = false;

        protected override Notification Init()
        {
            return new Notification();
        }
    }
}
